lib/kramdown/parser/kramdown/codespan.rb

```diff
@@ -25,8 +25,18 @@ def parse_codespan
           return
         end
 
-        if (text = @src.scan_until(/#{result}/))
-          text.sub!(/#{result}\Z/, '')
+        # assign static regex to avoid allocating the same on every instance
+        # where +result+ equals a single-backtick. Interpolate otherwise.
+        if result == '`'
+          scan_pattern = /`/
+          str_sub_pattern = /`\Z/
+        else
+          scan_pattern = /#{result}/
+          str_sub_pattern = /#{result}\Z/
+        end
+
+        if (text = @src.scan_until(scan_pattern))
+          text.sub!(str_sub_pattern, '')
```

Comment:
At least for this line we can do even better by using `text.chomp!(result)` instead of `#sub!`, and don't need a regexp at all. Then the `if` clause can be inlined as ternary operator in the line above this one.

Reply:
That's a splendid suggestion, Thomas! :clap: :clap: